### PR TITLE
feat: add server-side pagination, sort and search to groups component

### PR DESCRIPTION
**Pull request by punwai**
## Description
- Add server-side pagination, sort and search to groups component
- Refactor code to conform to current guidelines

## Related issues and discussion
#234

## Screenshots, if any
groups

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by punwai** [comment]
I have not added a search for the externalId, because it seems like the API endpoint does not work. 
I am not sure if I am doing anything wrong, but here is a sample queried I tried on the website which returns an empty array.

https://demo.openmf.org/fineract-provider/api/v1/groups?tenantIdentifier=default&pretty=true&externalId=009

File: src/app/groups/groups.service.ts [hunk 1]
```diff
@@ -1,17 +1,45 @@
...
+    let httpParams = new HttpParams()
+      .set('offset', offset.toString())
+      .set('limit', limit.toString())
+      .set('sortOrder', sortOrder)
+      .set('orderBy', orderBy)
+      .set('paged', 'true');
+    // filterBy: name
+    filterBy.forEach(function (filter: any) {
+      if (filter.value) {
+        console.log(filter.value);
```
Comment (abhaychawla) [inline src/app/groups/groups.service.ts]: Remove any `console.log` statements.
```suggestion
```

**Comment by abhaychawla** [comment]
> I have not added a search for the externalId, because it seems like the API endpoint does not work.
> I am not sure if I am doing anything wrong, but here is a sample queried I tried on the website which returns an empty array.
> 
> https://demo.openmf.org/fineract-provider/api/v1/groups?tenantIdentifier=default&pretty=true&externalId=009

Raise a Fineract ticket and a corresponding issue linked to it here for this.

**Comment by abhaychawla** [comment]
Contributor is inactive, but merging the PR as the required changes do not impact the functionality and can be done as part of refactoring later. Updating the commit message manually before merge to be in sync with the guidelines.

**Pull request merged**